mac/Keyman4MacIM/Keyman4MacIM/KMInputController.m

```diff
@@ -59,102 +61,44 @@ - (void)handleBackspace:(NSEvent *)event {
+  os_log_debug([KMLogs lifecycleLog], "***KMInputController inputMethodChangedClient, deactivating old eventHandler and activating new one");
+  if (_eventHandler != nil) {
+    [_eventHandler deactivate];
   }
-}
+  _eventHandler = [[KMInputMethodEventHandler alloc] initWithClient:[KMInputMethodLifecycle getClientApplicationId] client:self.client];
 
-- (void)timerAction:(id)lastServer {
-  @synchronized(servers) {
-    if (servers.count == 0) {
-      if (_eventHandler != nil) {
-        [_eventHandler deactivate];
-        _eventHandler = nil;
-      }
-      [self.AppDelegate sleepFollowingDeactivationOfServer:lastServer];
-    }
-  }
 }
 
+- (void)activateServer:(id)sender {
+  [sender overrideKeyboardWithKeyboardNamed:@"com.apple.keylayout.US"];
+  [KMInputMethodLifecycle.shared activateClient:sender];
+}
 
-/*
- - (NSDictionary *)modes:(id)sender {
- if ([self.AppDelegate debugMode])
- os_log_debug([KMLogs lifecycleLog], "*** Modes ***");
- if (_kmModes == nil) {
- NSDictionary *amhMode = [[NSDictionary alloc] initWithObjectsAndKeys:@"keyman.png", kTSInputModeAlternateMenuIconFileKey,
- [NSNumber numberWithBool:YES], kTSInputModeDefaultStateKey,
- [NSNumber numberWithBool:YES], kTSInputModeIsVisibleKey,
- @"A", kTSInputModeKeyEquivalentKey,
- [NSNumber numberWithInteger:4608], kTSInputModeKeyEquivalentModifiersKey,
- [NSNumber numberWithBool:YES], kTSInputModeDefaultStateKey,
- @"keyman.png", kTSInputModeMenuIconFileKey,
- @"keyman.png", kTSInputModePaletteIconFileKey,
- [NSNumber numberWithBool:YES], kTSInputModePrimaryInScriptKey,
- @"smUnicodeScript", kTSInputModeScriptKey,
- @"amh", @"TISIntendedLanguage", nil];
- 
- NSDictionary *hinMode = [[NSDictionary alloc] initWithObjectsAndKeys:@"keyman.png", kTSInputModeAlternateMenuIconFileKey,
- [NSNumber numberWithBool:YES], kTSInputModeDefaultStateKey,
- [NSNumber numberWithBool:YES], kTSInputModeIsVisibleKey,
- @"H", kTSInputModeKeyEquivalentKey,
- [NSNumber numberWithInteger:4608], kTSInputModeKeyEquivalentModifiersKey,
- [NSNumber numberWithBool:YES], kTSInputModeDefaultStateKey,
- @"keyman.png", kTSInputModeMenuIconFileKey,
- @"keyman.png", kTSInputModePaletteIconFileKey,
- [NSNumber numberWithBool:YES], kTSInputModePrimaryInScriptKey,
- @"smUnicodeScript", kTSInputModeScriptKey,
- @"hin", @"TISIntendedLanguage", nil];
- 
- NSDictionary *modeList = [[NSDictionary alloc] initWithObjectsAndKeys:amhMode, @"com.apple.inputmethod.amh", hinMode, @"com.apple.inputmethod.hin", nil];
- NSArray *modeOrder = [[NSArray alloc] initWithObjects:@"com.apple.inputmethod.amh", @"com.apple.inputmethod.hin", nil];
- _kmModes = [[NSDictionary alloc] initWithObjectsAndKeys:modeList, kTSInputModeListKey,
- modeOrder, kTSVisibleInputModeOrderedArrayKey, nil];
- }
- 
- return _kmModes;
- }
- */
+- (void)deactivateServer:(id)sender {
+  [KMInputMethodLifecycle.shared deactivateClient:sender];
+  [[NSNotificationCenter defaultCenter] removeObserver:self];
```

Comment:
This doesn't seem to have a corresponding addObserver in activateServer? Perhaps a comment may clarify